### PR TITLE
HSOpticalFlow SYCL migration

### PR DESCRIPTION
**Pull request by ShwethaSelma**
Signed-off-by: ShwethaSelma <shwethasshetty27@gmail.com>

# Adding a New Sample(s)

## Description
HSOpticalFlow Estimation Application Migration from CUDA to SYCL.

## Checklist
Administrative
- [X] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): <Choudary, Chekuri; Krishnappa, Rakshith>
- [ ] If you have any new dependencies/binaries, inform the oneAPI Code Samples Project Manager

Code Development
- [X] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [X] Adhere to readme template 
- [X] Enforce format via clang-format config file
- [X] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [X] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [X] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com

Security and Legal
- [ ] OSPDT Approval (see Project Manager for assistance)
- [X] Compile using the following compiler flags and fix any warnings, the falgs are: "/Wall -Wformat-security -Werror=format-security"
- [ ] Bandit Scans (Python only)
- [ ] Virus scan

Review
- [X] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [X] Review readme with Tom Lenth(@tomlenth) and/or Project Manager
- [X] Tested using Dev Cloud when applicable

### CODE REVIEW

File: DirectProgramming/DPC++/StructuredGrids/guided_HSOpticalflow_SYCLMigration/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+### Run the `HSOpticalFlow` Sample in Intel&reg; DevCloud
+
+When running a sample in the Intel&reg; DevCloud, you must specify the compute node (CPU, GPU, FPGA) and whether to run in batch or interactive mode. For more information, see the Intel&reg; oneAPI Base Toolkit [Get Started Guide](https://devcloud.intel.com/oneapi/get_started/).
+
+#### Build and Run Samples in Batch Mode (Optional)
+
+You can submit build and run jobs through a Portable Bash Script (PBS). A job is a script that submitted to PBS through the `qsub` utility. By default, the `qsub` utility does not inherit the current environment variables or your current working directory, so you might need to submit jobs to configure the environment variables. To indicate the correct working directory, you can use either absolute paths or pass the `-d \<dir\>` option to `qsub`.
+
+1. Open a terminal on a Linux* system.
+2. Log in to Intel&reg;7 DevCloud.
```
Comment (krisrak) [inline DirectProgramming/DPC++/StructuredGrids/guided_HSOpticalflow_SYCLMigration/README.md:117]: remove "7"

**Review by krisrak — CHANGES_REQUESTED** [review]
also remove "a.out" in 02* and 03* folders

File: DirectProgramming/DPC++/StructuredGrids/guided_HSOpticalflow_SYCLMigration/README.md (hunk 1 above)
Comment (ShwethaSelma) [reply DirectProgramming/DPC++/StructuredGrids/guided_HSOpticalflow_SYCLMigration/README.md]: Deleted a.out files and removed "7" from Readme.

**Review by krisrak — APPROVED** [review]
Looks good, can be merged

**Review by praveenkk123 — APPROVED** [review]

**Pull request merged**